### PR TITLE
proto: folder structure adhere to buf

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

This PR makes the proto structure adhere to the recommended structure in buf: https://buf.build/docs/lint-checkers#file_layout

todo: 
- [x] update ADR
- [x] add changelog & upgrading.md sections

Closes: #5021

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -31,6 +31,16 @@ amino to proto3 encoding and breaking changes to the header.
...
 
+### Protobuf
+
+With this release we are happy to announce the full protobuf migration of the Tendermint repo. This consists of changes that you may need to be aware of:
+
+- All proto files have been moved under one directory, `/proto`. This is in line with the recommended file layout by [buf](https://buf.build), you can read more about it [here](https://buf.build/docs/lint-checkers#file_layout)
+- We use the generated protobuf types for only on disk and over the wire serialization. This means that these changes should not effect you as user of Tendermint.
+- A few notable changes in the abci:
+  - In `ValidatorUpdates` the public key type has been migrated to a protobuf `oneof` type. Since Tendermint only supports ed25519 validator keys this is the only available key in the oneof.
+  - AppStateBytes is protobuf encoded instead of amino encoded <!-- TODO: see if this was even amino encoded before-->
```
Comment (marbar3778) [inline UPGRADING.md:42]: Have been reading this spec and AppStateBytes is mentioned as being amino encoded. Was this actually the case tho? 

This means that applications built in any lang would need amino to work??? right

**Comment by marbar3778** [comment]
marking as R4r but have a question in relation to appstatebytes

**Comment by clevinson** [comment]
@marbar3778 in cosmos-sdk, we only have the `.proto` files in the proto folder, and no generated go code. Generated code does not need to live in those folders to adhere to buf conventions. Maybe there are other reasons this has been done on the TM side.. but just wanted to flag.

**Comment by marbar3778** [comment]
We use proto files differently than in sdk. We only use the generated types for disk and on wire encoding. We could move them somewhere else but don't see a benefit for it living elsewhere compared to here. Do you know if this would be a negative?

File: UPGRADING.md (hunk 1 above)
Comment (melekes) [reply UPGRADING.md:42]: amino-encoded JSON (https://github.com/tendermint/tendermint/blob/95d2d136cdcff2a3bb29d4e4271d757e4f88867c/consensus/replay.go#L311)

> This means that applications built in any lang would need amino to work??? right

no, they will need a JSON parser

Comment (marbar3778) [reply UPGRADING.md:42]: hmm the sdk is talking about supporting both proto and amino encoded json for genesis, but based off this its not possible..

Comment (melekes) [reply UPGRADING.md:42]: note: Tendermint only initially sends json-encoded app_state from genesis.json to the app. at no point after that Tendermint uses AppStateBytes.

Comment (marbar3778) [reply UPGRADING.md:42]: ah okay, thank you for that

File: .gitignore [hunk 2]
```diff
@@ -43,6 +43,4 @@ terraform.tfstate
 terraform.tfstate.backup
 terraform.tfstate.d
 
-.vscode
```
Comment (marbar3778) [inline .gitignore:46]: it is still on line 30

**Pull request merged**